lib/exhal/assertions.ex

```diff
@@ -67,6 +67,13 @@ defmodule ExHal.Assertions do
   end
 
   @doc"""
+    Checks if a link is present in a rendered document.
+  """
+  def has_link?(json_map, rel) do
+    Kernel.get_in(json_map, ["_links", rel])
```

Comment:
This brought up interesting questions about the roles of Transcoder and Document. Since the use case are interested in is using a transcoder with `deflink` to define a link, but when the map we pass the the transcoder lacks that link name, we want the rendered json to not include that link, it seems like the work should be done in the transcoder. I'm open to persuasion though, as I think there are nuances about when and where empty links should/should not be kept around for internal representations.